include/vinecopulib/vinecop/implementation/tools_select.ipp

```diff
@@ -546,25 +567,78 @@ FamilySelector::finalize(size_t trunc_lvl)
 //! @param tree a vine tree.
 //! @return The pseudo-observations for the pair coula, extracted from
 //!     the h-functions calculated in the previous tree.
-inline Eigen::MatrixXd
-VinecopSelector::get_pc_data(size_t v0, size_t v1, const VineTree& tree)
+inline void
+VinecopSelector::add_pc_info(const EdgeIterator& e, VineTree& tree)
```

Comment:
Why is this better?

Reply:
Also, there seem to be a lot of code replication, wouldn't there be a way to modularize a bit?

Reply:
My bad, I saw the old function below :) Still, my comment about modularization stands.

Reply:
Done